src/cdk-experimental/scrollable-table-body/scrollable-table-body-layout.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Inject, Injectable, Input} from '@angular/core';
+import {CdkTable} from '@angular/cdk/table/table';
+import {DOCUMENT} from '@angular/common';
+import {
+  _TABLE_LAYOUT_STRATEGY,
+  _TableLayoutStrategy,
+  _DefaultTableLayoutStrategy,
+} from '@angular/cdk/table/table-layout-strategy';
+
+/**
+ * A {@link _TableLayoutStrategy} that enables scrollable body content for flex tables.
+ */
+@Injectable()
+class ScrollableTableBodyLayoutStrategy implements _TableLayoutStrategy {
+  private defaultLayout: _DefaultTableLayoutStrategy;
+  private _pendingMaxHeight = 'none';
+  private _scrollViewport?: HTMLElement;
+  readonly headerCssClass = 'cdk-table-scrollable-table-header';
+  readonly bodyCssClass = 'cdk-table-scrollable-table-body';
+  readonly footerCssClass = 'cdk-table-scrollable-table-footer';
+
+  constructor(@Inject(DOCUMENT) private readonly _document: any) {
+    this.defaultLayout = new _DefaultTableLayoutStrategy(this._document);
+  }
+
+  /**
+   * Returns the DOM structure for a native table. Scrollable body content is not supported for
+   * native tables. Return `null` to use the default {@link CdkTable} native table layout.
```

Comment:
Maybe you discussed this with @andrewseguin and @mmalerba, but what's the background on this only supporting flexbox-based table? We want to strongly encourage people to use the native table, so not supporting that for virtual-scroll seems like a problem to me.